lib/handsoap/service.rb

```diff
@@ -265,7 +269,9 @@ def dispatch(doc, action)
         fire_on_log_header { "HandSoap Response [#{dispatch_id}]: length: [#{@http_client.body_str.length}], HTTP-Status: [#{@http_client.response_code}], Content-Type: [#{@http_client.content_type}]" }
         fire_on_log_body { Handsoap.pretty_format_envelope(@http_client.body_str) }
 
-        soap_response = Response.new(@http_client.body_str, envelope_namespace)
+        response = OpenStruct.new(:content => @http_client.body_str, :headers => @http_client.headers)
```

Comment:
I think using `Struct` here might have less overhead per request. You could use it to define a `CurbResponse` class and use that in place of `OpenStruct`.